src/library/searchquery.cpp

```diff
@@ -324,13 +331,34 @@ bool NumericFilterNode::match(const TrackPointer& pTrack) const {
             continue;
         }
 
-        double dValue = value.toDouble();
+        const double dValue = value.toDouble();
+
+        // We are reducing the precision of double since an operation on Beats: getBpm
+        // has changed recently. It is dynamically calculated instead of using a stored
+        // precise value thus leading to a slight double precision error on comparison
+        // operations.
```

Comment:
```suggestion
        // Reduce the precision of the doubles because an operation on Beats::getBpm
        // is dynamically calculated instead of using a using precise value, thus there is
        // a slight double precision error on comparison operations.
```
Don't use relative time terms like "recently" in code comments. If that comment is still there in 10 years, it would be confusing.